kmath-jafama/src/main/kotlin/space/kscience/kmath/jafama/CmnFastMath.java

```diff
@@ -0,0 +1,2112 @@
+                    // Not using qZero anymore so not updating it.
+                    twoPowQZero *= TWO_POW_24;
+                    iq5 = (int)fw;
+                } else { // jz == jk+1 == 5
+                    // Case not encountered in tests, but still handling it.
+                    // Would require use of iq6, with jz = 6.
+                    return jdkRemainderPiO2(angle, negateRem);
+                }
+            } else {
+                if (jz == jk) {
+                    iq4 = (int)z;
+                } else { // jz == jk+1 == 5
+                    // Case not encountered in tests, but still handling it.
+                    iq5 = (int)z;
+                }
+            }
+        }
+
+        fw = twoPowQZero;
+
+        if (jz == 5) {
+            q5 = fw*(double)iq5;
+            fw *= TWO_POW_N24;
+        } else {
+            q5 = 0.0;
+        }
+        q4 = fw*(double)iq4;
+        fw *= TWO_POW_N24;
+        q3 = fw*(double)iq3;
+        fw *= TWO_POW_N24;
+        q2 = fw*(double)iq2;
+        fw *= TWO_POW_N24;
+        q1 = fw*(double)iq1;
+        fw *= TWO_POW_N24;
+        q0 = fw*(double)iq0;
+
+        /*
+         * We just use HI part of the result.
+         */
+
+        fw = PIO2_TAB0*q5;
+        fw += PIO2_TAB0*q4 + PIO2_TAB1*q5;
+        fw += PIO2_TAB0*q3 + PIO2_TAB1*q4 + PIO2_TAB2*q5;
+        fw += PIO2_TAB0*q2 + PIO2_TAB1*q3 + PIO2_TAB2*q4 + PIO2_TAB3*q5;
+        fw += PIO2_TAB0*q1 + PIO2_TAB1*q2 + PIO2_TAB2*q3 + PIO2_TAB3*q4 + PIO2_TAB4*q5;
+        fw += PIO2_TAB0*q0 + PIO2_TAB1*q1 + PIO2_TAB2*q2 + PIO2_TAB3*q3 + PIO2_TAB4*q4 + PIO2_TAB5*q5;
+
+        if ((ih != 0) ^ negateRem) {
+            fw = -fw;
+        }
+
+        return encodeRemainderAndQuadrant(fw, n&3);
+    }
+
+    //--------------------------------------------------------------------------
+    // PRIVATE METHODS
+    //--------------------------------------------------------------------------
+
+    /**
+     * Redefined here, to avoid cyclic dependency with (Strict)FastMath.
+     *
+     * @param value A double value.
+     * @return -1 if sign bit is 1, 1 if sign bit is 0.
+     */
+    private static long signFromBit_antiCyclic(double value) {
+        // Returning a long, to avoid useless cast into int.
+        return ((Double.doubleToRawLongBits(value)>>62)|1);
+    }
+
+    private static boolean getBooleanProperty(
+            final String key,
+            boolean defaultValue) {
+        final String tmp = System.getProperty(key);
+        if (tmp != null) {
+            return Boolean.parseBoolean(tmp);
+        } else {
+            return defaultValue;
+        }
+    }
+
+    /**
+     * Use look-up tables size power through this method,
+     * to make sure is it small in case java.lang.Math
+     * is directly used.
+     */
+    private static int getTabSizePower(int tabSizePower) {
+        return (FM_USE_JDK_MATH && SFM_USE_JDK_MATH) ? Math.min(2, tabSizePower) : tabSizePower;
+    }
+}
```

Comment:
Remove it and add jafama as Gradle dependency. You can use Maven coordinates net.jafama:jafama:2.3.2

Reply:
You also should remove all the other codes copied from the Jafama repository.